Xamarin.Essentials/FileSystem/FileSystem.ios.cs

```diff
@@ -176,25 +179,62 @@ internal override Task<Stream> PlatformOpenReadAsync()
         }
     }
 
+    // TODO: Think if we should rename it as UIImage is just an input
     class UIImageFileResult : FileResult
     {
-        readonly UIImage uiImage;
-        NSData data;
-
-        internal UIImageFileResult(UIImage image)
+        internal UIImageFileResult(UIImage image, NSDictionary metadata)
             : base()
         {
-            uiImage = image;
-
-            FullPath = Guid.NewGuid().ToString() + FileSystem.Extensions.Png;
-            FileName = FullPath;
+            FileName = Guid.NewGuid() + FileSystem.Extensions.Jpg;
+            var tempDir = NSFileManager.DefaultManager.GetTemporaryDirectory().Path;
+            FullPath = Path.Combine(tempDir, FileName);
+            SaveImageWithMetadata(image, metadata);
```

Comment:
Since this is now a more complex ctor that does a fair bit of work, maybe this should really be transformed as your comment suggests. Rename this to `ImageFileResult` and then use a ctor that takes a path. Then maybe add a static method such as `Create` and give an indication that work is being done. Or since this is now not doing anything, just move the logic to a method on FileSystem and use that to get a path to create a plain old FileResult?

Reply:
In fact, it might just be work reverting this class and keeping it that simple logic, and then moving your code to FileSystem. We might not use this class, but we can remove it later as it still might have a bit of value now for other cases.

In the actual code, just call the new method to get a string path, and return the new FileResult(path). This is what we do for Android anyway.

Reply:
> Or since this is now not doing anything, just move the logic to a method on FileSystem and use that to get a path to create a plain old FileResult?

This Class is for creating files from camera. There is no path at the moment there (I think camera creates just UIImage when taking images, unlike with videos, when it creates the file in the temp dir)

Reply:
> In fact, it might just be work reverting this class and keeping it that simple logic, and then moving your code to FileSystem. We might not use this class, but we can remove it later as it still might have a bit of value now for other cases.
> 
> In the actual code, just call the new method to get a string path, and return the new FileResult(path). This is what we do for Android anyway.

I think about this new implementation as a new replacement of current way of saving images. Current implementation does not really do what it should. Check the list if bugs which this PR suppose to fix.